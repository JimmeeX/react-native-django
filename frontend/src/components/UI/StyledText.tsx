import React from 'react';
import { Text, StyleSheet } from 'react-native';

interface StyledTextProps {
  children: React.ReactNode;
}

const StyledText: React.FC<StyledTextProps> = ({ children }) => {
  return <Text style={styles.text}>{children}</Text>;
};

const styles = StyleSheet.create({
  text: { fontFamily: 'open-sans' }
});

export default StyledText;
