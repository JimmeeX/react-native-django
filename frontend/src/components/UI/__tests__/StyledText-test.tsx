import * as React from 'react';
import renderer from 'react-test-renderer';

import StyledText from '../StyledText';

describe('StyledText Component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<StyledText>Snapshot test</StyledText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
