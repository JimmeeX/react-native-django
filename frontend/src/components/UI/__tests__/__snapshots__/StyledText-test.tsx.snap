// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText Component matches snapshot 1`] = `
<Text
  style={
    Object {
      "fontFamily": "open-sans",
    }
  }
>
  Snapshot test
</Text>
`;
